libcontainer/network_linux.go

```diff
@@ -171,7 +171,7 @@ func (v *veth) create(n *network, nspid int) (err error) {
 			netlink.LinkDel(veth)
 		}
 	}()
-	if err := v.attach(&n.Network); err != nil {
+	if err = v.attach(&n.Network); err != nil {
```

Comment:
I'm confused, what did this actually fix? Is there a reason we are setting `err` in the parent scope to `nil` in this round-about way?